src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -285,14 +287,66 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: E
     )
   }
 
-  def find(agoraSearch: AgoraEntity,
-           agoraProjection: Option[AgoraEntityProjection],
-           entityTypes: Seq[AgoraEntityType.EntityType],
-           username: String): Future[Seq[AgoraEntity]] = {
+  def listDefinitions(username: String): Future[Seq[MethodDefinition]] = {
+
+    val methodsFuture = findWithIds(AgoraEntity(), None, Seq(AgoraEntityType.Workflow), username)
+    val configsFuture = findWithIds(AgoraEntity(), None, Seq(AgoraEntityType.Configuration), username)
+
+    configsFuture flatMap { configSnapshots =>
+      methodsFuture flatMap { methodSnapshots =>
+        // group/count config snapshots by methodId
+        val configCounts:Map[Option[ObjectId],Int] = configSnapshots
+            .groupBy(_.methodId)
+            .map { kv => kv._1 -> kv._2.size}
+
+        // find public-ness
+        permissionsDataSource.inTransaction { db =>
+          db.aePerms.listPublicAliases flatMap { publicAliases =>
+
+            // apply public status
+            val snapshotsWithPublic = methodSnapshots.map { snap =>
+              val snapshotAlias = permissionsDataSource.dataAccess.aePerms.alias(snap)
+              snap.copy(public = Some(publicAliases.contains(snapshotAlias)))
+            }
+
+            // find and add owners
+            db.aePerms.listOwnersAndAliases map { ownersAndAliases =>
+
+              val annotatedSnapshots = snapshotsWithPublic.map { snap =>
+                val snapshotAlias = permissionsDataSource.dataAccess.aePerms.alias(snap)
+                val owners = ownersAndAliases.collect {
+                  case ((alias:String,email:String)) if alias == snapshotAlias => email
+                }
+                snap.addManagers(owners)
+              }
+
+              // group method snapshots by namespace/name; count method snapshots and sum config counts
+              val groupedMethods = annotatedSnapshots.groupBy( ae => (ae.namespace,ae.name))
+
+              groupedMethods.values.map { aes:Seq[AgoraEntity] =>
+                val numSnapshots:Int = aes.size
+                val numConfigurations:Int = aes.map { ae => configCounts.getOrElse(ae.id, 0) }.sum
+                val isPublic = aes.exists(_.public.contains(true))
+                val managers = aes.flatMap { ae => ae.managers }
+
+                // use the most recent (i.e. highest snapshot value) to populate the definition
+                val latestSnapshot = aes.maxBy(_.snapshotId.getOrElse(Int.MinValue))
+                MethodDefinition(latestSnapshot, managers, isPublic, numConfigurations, numSnapshots)
+              }.toSeq
+            }
+          }
```

Comment:
it seems like this could be done using a for comprehension to remove some of the nested maps/flatmaps.